client/executor/src/wasm_runtime.rs

```diff
@@ -148,15 +154,20 @@ const MAX_RUNTIMES: usize = 2;
 /// values of mutable globals. Follow-up requests to fetch a runtime return this one instance with
 /// the memory reset to the initial memory. So, one runtime instance is reused for every fetch
 /// request.
-///
-/// The size of cache is equal to `MAX_RUNTIMES`.
 pub struct RuntimeCache {
 	/// A cache of runtimes along with metadata.
 	///
 	/// Runtimes sorted by recent usage. The most recently used is at the front.
-	runtimes: Mutex<[Option<Arc<VersionedRuntime>>; MAX_RUNTIMES]>,
+	///
+	/// These runtime instances are only used in consensus code context and their count is limited
+	/// by `max_consensus_runtime_instances`.
+	consensus_runtimes: Mutex<[Option<Arc<VersionedRuntime>>; MAX_RUNTIMES]>,
 	/// The size of the instances cache for each runtime.
-	max_runtime_instances: usize,
+	max_consensus_runtime_instances: usize,
+	/// Same as `consensus_runtimes`, but used in offchain code context. Their size is always bound
+	/// by 1 (can easily be configurable, but we don't need this right now).
+	offchain_runtimes: Mutex<[Option<Arc<VersionedRuntime>>; MAX_RUNTIMES]>,
```

Comment:
I guess offchain is still using same wasm as the runtime.
So we can use twice as much memory of instantiated wasm now, I guess it is fine.

Reply:
Could it make sense to have MAX_RUNTIMES_OFFCHAIN. Did we consider only one with two instance?

Reply:
Afterthought: one instance would be really bad. I'm online being potentially stuck behind phragmen.